substrafl/dependency/path_management.py

```diff
@@ -0,0 +1,201 @@
+        excluded_regex: Optional[List[str]] = None,
+    ) -> List[PosixPath]:
+        """Copy paths from `src` to `dest_dir` respecting exlclusion/non-exclusion paths provided through `excluded_regex`,
+            `excluded`, `not_excluded`.
+
+        Args:
+            dest_dir (PosixPath): Directory where the file are going to be copied into.
+            src (List[PosixPath]): Path to copy
+            not_excluded (Optional[List[PosixPath]], optional): Paths to remove from the paths found in
+                `excluded`/`not_excluded`.
+                Defaults to None.
+            excluded (Optional[List[PosixPath]], optional): Paths to exclude from the `src` during the copy.
+                Defaults to None.
+            excluded_regex (Optional[List[str]], optional): Regex to find paths in `src` that will be excluded.
+                Always includes common data formats (see substrafl.dependency.EXCLUDED_PATHS_REGEX_DEFAULT)
+                Defaults to None.
+
+        Returns:
+            List[PosixPath]: Copied paths.
+        """
+        raise NotImplementedError("Abstract class do not provide implementation for this method")
+
+
+class DependencyPathManagement(BaseDependencyPathManagement):
+    @classmethod
+    def copy_paths(
+        cls,
+        *,
+        dest_dir: PosixPath,
+        src: List[PosixPath],
+        not_excluded: Optional[List[PosixPath]] = None,
+        excluded: Optional[List[PosixPath]] = None,
+        excluded_regex: Optional[List[str]] = None,
+    ) -> List[PosixPath]:
+        """Copy paths from `src` to `dest_dir` respecting exlclusion/non-exclusion paths provided through `excluded_regex`,
+            `excluded`, `not_excluded`
+
+        Args:
+            dest_dir (PosixPath): Directory where the file are going to be copied into
+            src (List[PosixPath]): Path to copy
+            not_excluded (Optional[List[PosixPath]], optional): Paths to remove from the paths found in
+                `excluded`/`not_excluded`.
+                Defaults to None.
+            excluded (Optional[List[PosixPath]], optional): Paths to exclude from the `src` during the copy.
+                Defaults to None.
+            excluded_regex (Optional[List[str]], optional): Regex to find paths in `src` that will be excluded.
+                Always includes common data formats (see substrafl.dependency.EXCLUDED_PATHS_REGEX_DEFAULT)
+                Defaults to None.
+
+        Raises:
+            ValueError: `dest_dir` is a file.
+            ValueError: One of the paths in `src` does not exist.
+
+        Returns:
+            List[PosixPath]: Copied paths.
+        """
+        if dest_dir.is_file():
+            raise ValueError(f"{dest_dir=} is a file. Cannot copy in a file.")
+
+        if not not_excluded:
+            not_excluded = []
+
+        if not excluded:
+            excluded = []
+
+        if not excluded_regex:
+            excluded_regex = []
```

Comment:
Shouldn't this be 

```suggestion
            excluded_regex = EXCLUDED_PATHS_REGEX_DEFAULT
```

Reply:
I currently set it in `get_excluded_paths`. note that your suggestion only add the default when nothing is provided. I remember discussing to activate it all the time, do I misremember ?

Reply:
yes makes sense, Friday evening reviews are definitely not the best ones 🙃